pkg/operator/apiservercontrollerset/apiservercontrollerset.go

```diff
@@ -0,0 +1,158 @@
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/apiservicecontroller"
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/nsfinalizercontroller"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/loglevel"
+	"github.com/openshift/library-go/pkg/operator/revisioncontroller"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/unsupportedconfigoverridescontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	apiregistrationv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
+	apiregistrationv1client "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/typed/apiregistration/v1"
+	apiregistrationinformers "k8s.io/kube-aggregator/pkg/client/informers/externalversions"
+)
+
+type APIServerControllerSet struct {
+	operatorClient v1helpers.OperatorClient
+	eventRecorder  events.Recorder
+
+	apiServiceController            *apiservicecontroller.APIServiceController
+	clusterOperatorStatusController *status.StatusSyncer
+	configUpgradableController      *unsupportedconfigoverridescontroller.UnsupportedConfigOverridesController
+	logLevelController              *loglevel.LogLevelController
+	finalizerController             *nsfinalizercontroller.FinalizerController
+	revisionController              *revisioncontroller.RevisionController
+}
+
+func NewAPIServerControllerSet(
+	operatorClient v1helpers.OperatorClient,
+	eventRecorder events.Recorder,
+) *APIServerControllerSet {
+	apiServerControllerSet := &APIServerControllerSet{
+		operatorClient: operatorClient,
+		eventRecorder:  eventRecorder,
+	}
+
+	return apiServerControllerSet
+}
+
+// WithConfigUpgradableController adds a controller for the operator to check for presence of
+// unsupported configuration and to set the Upgradable condition to false if it finds any
+func (cs *APIServerControllerSet) WithConfigUpgradableController() *APIServerControllerSet {
+	cs.configUpgradableController = unsupportedconfigoverridescontroller.NewUnsupportedConfigOverridesController(cs.operatorClient, cs.eventRecorder)
+	return cs
+}
+
+func (cs *APIServerControllerSet) WithoutConfigUpgradableController() *APIServerControllerSet {
+	cs.configUpgradableController = nil
+	return cs
+}
+
+// WithLogLevelController adds a controller that configures logging for the operator
+func (cs *APIServerControllerSet) WithLogLevelController() *APIServerControllerSet {
+	cs.logLevelController = loglevel.NewClusterOperatorLoggingController(cs.operatorClient, cs.eventRecorder)
+	return cs
+}
+
+func (cs *APIServerControllerSet) WithoutLogLevelController() *APIServerControllerSet {
+	cs.logLevelController = nil
```

Comment:
you will later need to know if this is an explicit choice to provide an error when someone doesn't handle  a new contorller.

Reply:
I wrapped the controllers so that I can check if they are unintentionally unset. The nsFinalizer struct is once again unimported thanks to that.